modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/rest/SW360AttachmentAwareClient.java

```diff
@@ -77,12 +82,54 @@ private T uploadAndAttachAttachment(T itemToModify, Path fileToAttach, HttpEntit
             Validate.validState(response.getBody() != null);
             return response.getBody();
         } catch (HttpClientErrorException | HttpServerErrorException e) {
-            LOGGER.warn("Request to get attach {} to {} failed with {}", fileToAttach, self, e.getStatusCode());
+            LOGGER.warn("Request to attach {} to {} failed with {}", fileToAttach, self, e.getStatusCode());
             LOGGER.debug("Error: ", e);
             return itemToModify;
         } catch (ExecutionException e) {
             LOGGER.warn("Request to attach {} to {} failed with {}", fileToAttach, self, e.getMessage());
             return itemToModify;
         }
     }
+
+    public List<SW360SparseAttachment> getReleaseAttachments(T itemToModify, HttpHeaders header) {
```

Comment:
I am just thinking whether this method makes sense at all. There might be a multitude of attachments and from an Antenna point of view, we should be interested in some of them. But typically we only need to download the sources nothing else. Or is it that the method only returns the references to the attachments as a list and I have to use the download method below for the real download?

By the way, does it make sense to make a _getReleaseAttachments_ method and then use a parameter T that allows to download any kind of attachment?

Reply:
The parameter T is for the HalResource. One gets an attachment of any object type (release, component or project).
Though, you are right. It is not named accordingly, leading to confusion. I'll check this again